src/bun.js/test/expect.zig

```diff
@@ -3776,6 +4101,98 @@ pub const ExpectStringMatching = struct {
     }
 };
 
+pub const ExpectCloseTo = struct {
+    pub usingnamespace JSC.Codegen.JSExpectCloseTo;
+
+    pub fn finalize(
+        this: *ExpectCloseTo,
+    ) callconv(.C) void {
+        VirtualMachine.get().allocator.destroy(this);
+    }
+
+    pub fn call(globalObject: *JSC.JSGlobalObject, callFrame: *JSC.CallFrame) callconv(.C) JSValue {
+        const args = callFrame.arguments(2).slice();
+
+        if (args.len == 0 or !args[0].isNumber()) {
+            globalObject.throwPretty("<d>expect.<r>closeTo<d>(<r>number<d>, precision?)<r>\n\nExpected a number value", .{});
+            return .zero;
+        }
+        const number_value = args[0];
+
+        var precision_value = if (args.len > 1) args[1] else JSValue.undefined;
+        if (precision_value.isUndefined()) {
+            precision_value = JSValue.jsNumberFromInt32(2); // default value from jest
+        }
+        if (!precision_value.isNumber()) {
+            globalObject.throwPretty("<d>expect.<r>closeTo<d>(number, <r>precision?<d>)<r>\n\nPrecision must be a number or undefined", .{});
+            return .zero;
+        }
+
+        if (Jest.runner.?.pending_test == null) {
+            const err = globalObject.createErrorInstance("expect.closeTo() must be called in a test", .{});
+            err.put(globalObject, ZigString.static("name"), ZigString.init("TestNotRunningError").toValueGC(globalObject));
+            globalObject.throwValue(err);
+            return .zero;
+        }
```

Comment:
We should fix this, this technically isn't right. Asymmetric matchers should be able to be defined outside the context of a test. For example:

```js
test.each([
  {
    actual: { a: 1, b: 2 },
    // this is not called in the test scope, but in global scope
    expected: { a: expect.closeTo(1), b: expect.closeTo(2) },
  }
])("%s", ({ actual, expected }) => {
  expect(actual).toEqual(expected);
});
```

I think we can just remove these checks for the matchers.

Reply:
I put it in because I saw the existing asymmetric matchers had this check, although I agree this is a good point

Reply:
Should I remove it from all asymmetric matchers?

Reply:
Yep